### PR TITLE
Finish check

### PR DESCRIPTION
**Pull request by shaunshapiro**
Displays on game page if either user is in check. However, the check method itself doesn't seem to work quite right, and I think it's due to the logic in can_move_with_capture? in models/piece.rb not taking into account how each piece moves, due to it not calling valid_move?...

### CODE REVIEW

**Comment by tzinser** [comment]
I see what your saying... My tests were failing locally when I was running rake and so I commented it out for testing since pawn valid_move was not implemented, and forgot to uncomment.  If you can uncomment that line and push it up with your changes?  Thanks

**Comment by shaunshapiro** [comment]
Ah ok... I noticed it was commented out and thought you did so because it brought up other problems, just was waiting to see what the reasoning was. I'll give it a try and see how it works

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -5,6 +5,13 @@ def show
     @pieces = @game.pieces
     @black_player = User.find_by(id: @game.black_uid)
     @white_player = User.find_by(id: @game.white_uid)
+    if @game.is_color_in_check?(0) && @game.is_color_in_check?(1)
```
Comment (johnnyiller) [inline app/controllers/games_controller.rb]: for readability you would want to use.

Piece::COLORS[:black]
Piece::COLORS[:white] 

to denote white and black.  Once a constant is defined you should use that constant consistently throughout the app.

File: app/controllers/pieces_controller.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ def update
     @piece = Piece.find(params[:id])
     @game = @piece.game
     @color = @piece.color
+    @last_move = Piece.order(:updated_at).last.color
```
Comment (johnnyiller) [inline app/controllers/pieces_controller.rb:12]: clever good use of the updated_at field

File: app/controllers/pieces_controller.rb [hunk 3]
```diff
@@ -16,12 +17,18 @@ def update
     	@color = 'black'
     end
 
+    if @last_move == 0
```
Comment (johnnyiller) [inline app/controllers/pieces_controller.rb]: again use 

Piece::COLORS[:black] == @last_move

alternatively you could implement a pair of methods in the pieces model and use those throughout the app.

def is_black?
  self.color == Piece::COLORS[:black]
end

def is_white?
  !is_black?
end

Just a suggestion

**Comment by johnnyiller** [comment]
I can't find anything wrong with the code that sticks out so if you want to merge I think it's safe but if you want to clean up those 1's and 0's that denote black and white then that might be a nice thing to do, if not no big deal, just merge the code to avoid blocking others.

**Comment by shaunshapiro** [comment]
Check logic still doesn't seem to work properly with that line uncommented in can_move_with_capture?

**Comment by shaunshapiro** [comment]
@johnnyiller should I still merge this branch now and we can fix the logic in another pull request, or leave it open for now? The stuff I added in this branch works, it's just the logic of the check method that might need a bit of work

**Comment by johnnyiller** [comment]
yeah I think you are good to merge this

**Pull request merged**